danny-avila/LibreChat
# 🗨️ feat: Prompts

### Summary

I implemented the initial features for the Prompt Library, enhancing prompt management capabilities as proposed in issue #3023. This includes functionalities for creating, using, and sharing prompts, as well as admin-specific controls and categorization.

- Implemented ability to create and use prompts easily.
- Added support for prompts to have "fill-in" variables.
- Created special variables for current date and current user.
- Enabled users to share prompts globally.
- Provided quick and easy access to prompts from the side panel.
- Added filtering functionality for prompts.
- Implemented advanced and optional versioning of prompts.
- Enabled categorization and labeling of prompts.
- Allowed admins to share prompts globally.
- Enabled admins to restrict user prompt creation.
- Allowed admins to add categories for prompts.

### Testing

To ensure the implemented features are working correctly:
 
1. Create a prompt and verify its usability within a chat.
2. Use fill-in variables within a prompt and confirm correct substitution.
3. Share a prompt globally and check its visibility to other users.
4. Access prompts from the side panel and verify quick access functionality.
5. Filter prompts and confirm accurate results.
6. Test versioning by creating multiple versions of a prompt.
7. Categorize prompts and check if they are listed under the correct labels.
8. Admins should try sharing and restricting user prompts and adding categories.

### Test Configuration:

- Test different user roles (admin, regular user).
- Ensure prompts work across various chat endpoints.
- Verify prompt management functionalities in different browsers.

### Checklist


Also closes  #3128
